python/xgbserver/xgbserver/__main__.py

```diff
@@ -44,10 +44,13 @@
         kserve.ModelServer().start([model])
     except ModelMissingError:
         logger.error(
-            f"fail to locate model file for model {args.model_name} under dir {args.model_dir},"
+            f"failed to locate model file for model {args.model_name} under dir {args.model_dir},"
             f"trying loading from model repository."
         )
-
+        # Case 1: Model will be loaded from model repository automatically, if present
+        # Case 2: In the event that the model repository is empty, it's possible that this is a scenario for
+        # multi-model serving. In such a case, models are loaded dynamically using the TrainedModel.
+        # Therefore, we start the server without any preloaded models
         kserve.ModelServer(
             registered_models=XGBoostModelRepository(args.model_dir, args.nthread)
-        ).start([model])
```

Comment:
Starting it without passing the model means that this method `XGBoostModelRepository` will load it, right?

In the other PR, there is a weird message: `Successfully copied /mnt/models to None`
which means that no model was loaded at all, right?

Reply:
The models will be loaded when the model repository is [instantiated](https://github.com/kserve/kserve/blob/d19e31040c558ef88774736f67a0c2af7dbc6bc2/python/xgbserver/xgbserver/xgboost_model_repository.py#L24). This log is misleading. This arises from [here](https://github.com/kserve/kserve/blob/d19e31040c558ef88774736f67a0c2af7dbc6bc2/python/xgbserver/xgbserver/model.py#L44). At this point the models are already downloaded to `/mnt/models` by the storage initializer init container. So calling `storage.download` with `/mnt/models` as the location leads to this misleading log. When the storage.download is called with `/mnt/models`, it just returns the the same path, doing nothing. I am not sure why we need to call `storage.download` in model server as it is already download by the storage initializer. 

@yuzisun

Reply:
That code probably can be removed, I think it is to ensure the model is in the canoncial location `/mnt/models` and make symlinks if it is downloaded in a different location.